### PR TITLE
Allow using an existing private registry secret

### PR DESCRIPTION
**Pull request by wilson1999112**
Hi @innobead,
The PR addresses issue #2157
There are 3 ways to install longhorn as mention in the doc: 
1. Using a Manifest File
2. Using a Helm Chart
3. Using a Rancher App

For 1., just type the name of the existing private registry secret.

For 2./3., modify `chart/values.yaml`.
For example in `chart/values.yaml`:
1. If you want to create a new secret for longhorn, enter your user/pass like this:
```
privateRegistry:
  createSecret: true
  registryUrl: <REGISTRY_URL>
  registryUser: <REGISTRY_USER>
  registryPasswd: <REGISTRY_PASSWORD>
  registrySecret: <REGISTRY_SECRET_NAME>
```
2. if you want to make use of the existing private registry secret, enter your user/pass like this:
```
privateRegistry:
  createSecret: false
  registryUrl: <REGISTRY_URL>
  registryUser: ~
  registryPasswd: ~
  registrySecret: <REGISTRY_SECRET_NAME>
```

### CODE REVIEW

**Comment by wilson1999112** [comment]
Hi @innobead,
The PR addresses issue #2157
There are 3 ways to install longhorn as mention in the doc: 
1. Using a Manifest File
2. Using a Helm Chart
3. Using a Rancher App

For 1., just type the name of the existing private registry secret.

For 2./3., modify `chart/values.yaml`.
For example in `chart/values.yaml`:
1. If you want to create a new secret for longhorn, enter your user/pass like this:
```
privateRegistry:
  secretCreation: true
  registryUrl: <REGISTRY_URL>
  registryUser: <REGISTRY_USER>
  registryPasswd: <REGISTRY_PASSWORD>
  registrySecret: <REGISTRY_SECRET_NAME>
```
2. if you want to make use of the existing private registry secret, enter your user/pass like this:
```
privateRegistry:
  secretCreation: false
  registryUrl: <REGISTRY_URL>
  registryUser: ~
  registryPasswd: ~
  registrySecret: <REGISTRY_SECRET_NAME>
```

**Review by jenting — CHANGES_REQUESTED** [review]
We need to update the `question.yaml` as well.

**Comment by wilson1999112** [comment]
> We need to update the `questions.yaml` as well.

addressed, thank you.

**Review by innobead — CHANGES_REQUESTED** [review]

File: chart/questions.yaml [hunk 1]
```diff
@@ -142,27 +142,33 @@ questions:
     type: string
     label: Longhorn CSI Driver Snapshotter Image Tag
     group: "Longhorn CSI Driver Images"
+- variable: privateRegistry.secretCreation
+  default: "true"
+  description: "Create a new private registry secret"
+  type: boolean
+  group: "Private Registry Settings"
+  label: Secret creation
```
[the remark this answers is not in the document]
Comment (wilson1999112) [reply chart/questions.yaml]: done!

File: chart/questions.yaml [hunk 2]
```diff
@@ -142,27 +142,33 @@ questions:
...
+  default: "true"
+  description: "Create a new private registry secret"
+  type: boolean
+  group: "Private Registry Settings"
+  label: Create Secret for Private Registry Settings
+  show_subquestion_if: true
+  subquestions:
+  - variable: privateRegistry.registryUser
+    label: Private registry user
+    description: "User used to authenticate to private registry"
```
Comment (jenting) [inline chart/questions.yaml:155]:
```suggestion
    description: "User used to authenticate to private registry."
```

File: chart/questions.yaml [hunk 3]
```diff
@@ -142,27 +142,33 @@ questions:
...
+    type: string
+    default: ""
+  - variable: privateRegistry.registryPasswd
+    label: Private registry password
+    description: "Password used to authenticate to private registry"
```
Comment (jenting) [inline chart/questions.yaml:160]:
```suggestion
    description: "Password used to authenticate to private registry."
```

File: chart/questions.yaml [hunk 4]
```diff
@@ -142,27 +142,33 @@ questions:
...
-- variable: privateRegistry.registryPasswd
-  label: Private registry password
-  description: "Password used to authenticate to private registry"
-  group: "Private Registry Settings"
-  type: password
-  default: ""
 - variable: privateRegistry.registrySecret
   label: Private registry secret name
-  description: "Longhorn will automatically generate a Kubernetes secret with this name and use it to pull images from your private registry."
+  description: "If create secret is true, create a Kubernetes secret with this name; else use the existing secret of this name. Use it to pull images from your private registry."
```
Comment (jenting) [inline chart/questions.yaml:171]:
```suggestion
  description: "If create a new private registry secret is true, create a Kubernetes secret with this name; else use the existing secret of this name. Use it to pull images from your private registry."
```

**Review by jenting — COMMENTED** [review]
We could consider moving lines 145-162 under line 170, which makes the UX more intuitive.
- WAS
image

- IS
image

**Review by jenting — APPROVED** [review]
I helped address the comment.

LGTM

**Review by innobead — APPROVED** [review]
LGTM

File: chart/questions.yaml [hunk 5]
```diff
@@ -148,24 +148,30 @@ questions:
...
-- variable: privateRegistry.registryPasswd
-  label: Private registry password
-  description: "Password used to authenticate to private registry"
-  group: "Private Registry Settings"
-  type: password
-  default: ""
 - variable: privateRegistry.registrySecret
   label: Private registry secret name
-  description: "Longhorn will automatically generate a Kubernetes secret with this name and use it to pull images from your private registry."
+  description: "If create a new private registry secret is true, create a Kubernetes secret with this name; else use the existing secret of this name. Use it to pull images from your private registry."
```
Comment (innobead) [inline chart/questions.yaml:153]: Ok this has been updated actually cc @c3y1huang @kaxing

**Pull request merged**